src/Poseidon/Analysis/CLI/FStats.hs

```diff
@@ -402,20 +405,64 @@ processBlocks statSpecs blocks = do
                 (estimateJackknife, stdErrJackknife) = computeJackknifeOriginal full_estimate block_weights partial_estimates
             in  return (full_estimate, estimateJackknife, stdErrJackknife)
 
+isRatio :: FStatType -> Bool
+isRatio F3star     = True
+isRatio FSTvanilla = True
+isRatio FST        = True
+isRatio _          = False
+
 -- outer list: stats, inner list: estimates per block
 processBlockIndividually :: [FStatSpec] -> BlockData -> [Double]
 processBlockIndividually statSpecs (BlockData _ _ _ allStatVals) = do
     (FStatSpec fType _ _, statVals) <- zip statSpecs allStatVals
-    case fType of
-        F3 ->
+    if isRatio fType then
             let numerator_value = statVals !! 0
                 numerator_norm = statVals !! 1
                 denominator_value = statVals !! 2
                 denominator_norm = statVals !! 3
                 num_full = numerator_value / numerator_norm
                 denom_full = denominator_value / denominator_norm
             in  return $ num_full / denom_full
-        _ ->
+        else
             let value = statVals !! 0
                 norm = statVals !! 1
             in  return $ value / norm
+
+checkForIllegalSampleSizes :: (MonadIO m) => PloidyVec -> EntityIndicesLookup -> FStatSpec -> m ()
```

Comment:
If so, then this should maybe be an update in the major version number.

Reply:
Hmm... well... depends on what you mean by "break". So before it was possible to run something like `F3(A, B; C)`, where `C` would be a group with a single haploid sample, and the result would be undefined (because you're the formula involves dividing by `n-1` which is then 0. This case is now caught and an error is thrown. So I guess on some level it breaks things, but that case before would have resulted in useless results.

It doesn't feel to me as if this warrants a major version update, to be honest. It's just a new sanity check that avoids people running illegal stats.